### PR TITLE
add options to disable and override ExplicitExpansion

### PR DESCRIPTION
**Pull request by Angelinsky7**
replace #4326
Pr to implement https://github.com/AutoMapper/AutoMapper/discussions/4325
from discussion from here : https://github.com/AutoMapper/AutoMapper/discussions/4323

### CODE REVIEW

**Comment by Angelinsky7** [comment]
@lbargaoanu i think i corrected almost everything. (apart from the override parameter that i really would like to talk about)

File: src/IntegrationTests/ExplicitExpansion/DisableExplicitExpansion.cs [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+    public class Context : LocalDbContext {
+        public DbSet<Entity> Entities { get; set; }
+    }
+    public class DatabaseInitializer : DropCreateDatabaseAlways<Context> {
+        protected override void Seed(Context context) {
+            context.Entities.Add(new() { Name = "Name" });
+            base.Seed(context);
+        }
+    }
+    protected override MapperConfiguration CreateConfiguration() => new(c => c.CreateProjection<Entity, Dto>().ForCtorParam("Name", o => o.ExplicitExpansion(false)));
```
Comment (lbargaoanu) [inline src/IntegrationTests/ExplicitExpansion/DisableExplicitExpansion.cs:102]: This should use mapping inheritance, the same as the test below.

Comment (Angelinsky7) [reply src/IntegrationTests/ExplicitExpansion/DisableExplicitExpansion.cs:102]: @lbargaoanu i added a new test for constructor mapping with inheritance (sorry for the long delay i needed to change my computer)

**Comment by Angelinsky7** [comment]
if you do that, you loose all the advantage of doing this pr !
i don't understand why you would want to make all user need to change all their code by needing to pass a value (it breaks compatibility)

**Comment by lbargaoanu** [comment]
Obviosly it doesn't, all the existing tests pass.

**Comment by Angelinsky7** [comment]
@lbargaoanu for the matter of this, i'm against this pr in this state...

**Comment by Angelinsky7** [comment]
@lbargaoanu yeah of course you removed all my tests :-)

**Comment by lbargaoanu** [comment]
Well, that's what's needed here. The `ForAllMembers` stuff is `ForAllOtherMembers` in disguise, nothing that makes sense for AM. If you want that, just add an `if` as I've already said.

**Comment by Angelinsky7** [comment]
i'm sorry but you didn't understand my usecase and why i was doing this pr... and your solution doesn't work.

**Comment by Angelinsky7** [comment]
but again, don't bother

**Pull request merged**